ReactWindows/ReactNative/Views/View/ReactViewManager.cs

```diff
@@ -0,0 +1,270 @@
+                            var props = await file.GetBasicPropertiesAsync();
+                            var type = file.ContentType;
+                            var path = drop ? "urn:future-access-list:" + Uri.EscapeUriString(GetAccessToken(file)) : "";
+
+                            files.Add(new JObject
+                            {
+                                { "name", file.Name },
+                                { "size", props.Size },
+                                { "type", type },
+                                { "uri", path }
+                            });
+
+                            items.Add(new JObject
+                            {
+                                { "kind", "file" },
+                                { "type", type }
+                            });
+
+                            types.Add(type);
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        System.Diagnostics.Debug.WriteLine("GetDataTransferInfo: " + ex);
+                    }
+                }
+            }
+
+            return new JObject
+            {
+                { "files", files },
+                { "items", items },
+                { "types", types }
+            };
+        }
+
+        private static string GetAccessToken(StorageFile file)
+        {
+            var futureAccessList = StorageApplicationPermissions.FutureAccessList;
+            var existingEntries = futureAccessList.Entries;
+            var capacity = futureAccessList.MaximumItemsAllowed;
+            var now = DateTime.Now;
+
+            // Use more aggresive timeout when number of items exceed 100 or half of the capacity
+            // Note: 100 is an arbitary limit in case that OS reduces the capacity (it is 1000 in RS1). 
+            var expiry = existingEntries.Count >= capacity / 2 || existingEntries.Count >= 100 ?
+                now.AddDays(-1) :
+                now.AddDays(-7);
+
+            foreach (var entry in existingEntries)
+            {
+                if (DateTime.TryParse(entry.Metadata, out DateTime time) && time < expiry)
+                {
+                    futureAccessList.Remove(entry.Token);
+                }
+            }
+
+            return futureAccessList.Add(file, now.ToString(System.Globalization.CultureInfo.InvariantCulture));
+        }
+
+        private static async void OnDragEnter(object sender, DragEventArgs args)
+        {
+            var view = sender as BorderedCanvas;
+            var data = await GetDataTransferInfo(args.DataView);
+
+            view.GetReactContext()
+                .GetNativeModule<UIManagerModule>()
+                .EventDispatcher
+                .DispatchEvent(new DragDropEvent(view.GetTag(), "topDragEnter", data));
+        }
+
+        private static async void OnDragOver(object sender, DragEventArgs args)
+        {
+            var view = sender as BorderedCanvas;
+
```

Comment:
Potential NullOperationException, either cast or do null check.

Reply:
I'm not sure if it can throw any exceptions here. The `as` operator doesn't throw exceptions, so the only option is if `sender` is not a `BorderedCanvas`, but how is that possible if the event handler was added to a `BorderedCanvas`?

Reply:
I guess you meant the bubbling mechanism: the drag/drop event might be coming from a nested not-a-BorderedCanvas element. I can add checks for that just in case.